zfs_diskio.c

```diff
@@ -113,7 +115,20 @@ DRESULT disk_ioctl(BYTE pdrv, BYTE cmd, void *buff)
 		if (disk_access_ioctl(pdrv_str[pdrv],
 				DISK_IOCTL_GET_SECTOR_COUNT, buff) != 0) {
 			ret = RES_ERROR;
-                }
+		}
+		break;
+
+	case GET_SECTOR_SIZE:
+		/* Zephyr's DISK_IOCTL_GET_SECTOR_SIZE returns sector size as a
+		 * 32-bit number while FatFS's GET_SECTOR_SIZE is supposed to
+		 * return a 16-bit number.
+		 */
+		if (disk_access_ioctl(pdrv_str[pdrv],
+				DISK_IOCTL_GET_SECTOR_SIZE, &sector_size) == 0) {
+			*(uint16_t *)buff = (uint16_t)sector_size;
```

Comment:
I think this requires some assertion or check of higher 16bits against 0.

Reply:
Agreed; add `(sector_size == (uint16_t)sector_size)` as a secondary condition.

Reply:
Added check. Thanks!